### PR TITLE
fix(workflows): detect+repair bazel#23880 poisoning, replace sick runner

### PR DESCRIPTION
**Pull request by gregmagolan**
## Problem

Bazel 8.x can leave its sandbox base in a state that crashes every subsequent invocation on the same `output_base`:

```
ERROR: Failed to initialize sandbox: fchmod (.../sandbox/linux-sandbox/<pid>/.../tmp...) (Operation not permitted)
FATAL: bazel crashed due to an internal error.
java.lang.IllegalStateException: Found unexpected entries in sandbox base. ... The entries are: linux-sandbox, _moved_trash_dir
    at SandboxModule.checkSandboxBaseTopOnlyContainsPersistentDirs(SandboxModule.java:573)
    at SandboxModule.afterCommand(SandboxModule.java:611)
```

When `LinuxSandboxedStrategy.create` throws `IOException` during sandbox setup, the Linux spawn runner never gets added to `SandboxModule.spawnRunners`. `afterCommand`'s cleanup iterates only registered runners, so the partial `<output_base>/sandbox/linux-sandbox/...` subtree survives on disk and the next bazel command hard-fails in `checkSandboxBaseTopOnlyContainsPersistentDirs`. The runner stays poisoned across jobs — every retry hits the same wedge. (`_moved_trash_dir` in the error message is a red herring; it's on Bazel's `SANDBOX_BASE_PERSISTENT_DIRS` whitelist. The offending entry is `linux-sandbox`.)

Fixed upstream by [`abe8d6090`](https://github.com/bazelbuild/bazel/commit/abe8d6090e9ddea898867cc59f7cb434079df4da) in Bazel 9.0+; not backported to any 8.x release (verified through 8.7.0). See bazelbuild/bazel#23880.

## Approach

Two new `BazelTrait` hooks the Workflows feature registers when `environment.runner != None`:

- **`bazel_attempt_end(ctx, exit_code)`** — fires after every bazel invocation (each retry attempt) and BEFORE the retry decision. The Workflows-registered handler calls a new `ctx.bazel.recover_poisoned_sandbox()` Rust helper that lists entries directly under `<output_base>/sandbox/` against Bazel's whitelist and `rm -rf`'s anything else. Outcomes:
  - `clean` — nothing to do (the common case).
  - `repaired` — entries were removed; the retry (or next job on this runner) starts from a healthy sandbox.
  - `still_poisoned` — at least one entry survived `rm -rf` (would need EPERM/EROFS on the rm itself). Runs `signal_instance_unhealthy` so the autoscaler replaces the instance.

- **`build_end(ctx, exit_code)`** — fires once with the final exit code (existing attribute; the new policy is a fresh hook). Signals unhealthy when the final code is `BLAZE_INTERNAL_ERROR (37)` or `LOCAL_ENVIRONMENTAL_ERROR (36)` — the two codes `default_retry` keys off, so a final retryable-class exit means every retry hit a server-internal error and the server is persistently sick. A retry sequence ending in a normal build/test failure (`1`/`3`/...) does NOT signal — the server produced a verdict, the runner is fine.

Two public dispatch helpers in `lib/bazel_runner.axl` — `dispatch_bazel_attempt_end` / `dispatch_bazel_build_end` — are called from every site that spawns a bazel build/test in built-in tasks: `build`, `test`, `run`, `gazelle`, `lint`, `format`, and all three `delivery` phases. The retry loop in `bazel_runner.axl` dispatches `bazel_attempt_end` between `invocation.wait()` and the retry decision so a successful repair lets the next attempt proceed.

Bazel 9.0+ is unaffected by the upstream bug; the mitigation is benign there — the sandbox dir is always clean and the detector is a single `readdir`.

---

### Changes are visible to end-users: yes

- Searched for relevant documentation and updated as needed: yes
- Breaking change (forces users to change their own code or config): no
- Suggested release notes appear below: yes

- Aspect Workflows runners now auto-recover from bazelbuild/bazel#23880-style sandbox poisoning across all built-in tasks (`build`, `test`, `run`, `gazelle`, `lint`, `format`, `delivery`). When `<output_base>/sandbox/` contains entries outside Bazel's whitelist after a bazel invocation, the offending subtrees are `rm -rf`'d so the next attempt (or next job on the runner) starts clean. If the removal fails, or if a bazel invocation exits with `BLAZE_INTERNAL_ERROR (37)` / `LOCAL_ENVIRONMENTAL_ERROR (36)`, the runner is marked unhealthy so the next job lands on a fresh instance.

### Test plan

- Rust unit tests in `engine::bazel::sandbox_recovery::tests` cover: empty/missing sandbox, whitelist-only entries (must survive), single `linux-sandbox` subtree repair, multiple per-strategy subtrees with `sandbox_stash` preserved alongside, `rm -rf` failure → `StillPoisoned`, and the three `output_base_from_flags` edge cases.
- AXL unit tests in `lib/sandbox_recovery_test.axl` (`aspect dev test-sandbox-recovery`) cover the attempt-end and build-end hook policies on clean / repaired / still_poisoned outcomes, retryable vs non-retryable exit codes, and the missing-signal-binary fallback.
- AXL unit tests in `lib/bazel_runner_test.axl` (`aspect dev test-bazel-runner`) pin the dispatcher contract: in-order invocation, exit-code passthrough, and that the two dispatchers are independently addressable.

### CODE REVIEW

File: crates/aspect-cli/src/builtins/aspect/bazel.axl [hunk 1]
```diff
@@ -81,5 +81,6 @@ BazelTrait = trait(
     build_start = attr(list[typing.Callable[[TaskContext], None]], default = [], description = "Hooks called once before the Bazel invocation starts"),
     build_event = attr(list[typing.Callable[[TaskContext, dict], None]], default = [], description = "Hooks called for each Build Event Protocol event received during the build"),
     build_retry = attr(typing.Callable[[int], bool], default = default_retry, description = "Predicate called with the exit code; return True to retry the build"),
-    build_end = attr(list[typing.Callable[[TaskContext, int], None]], default = [], description = "Hooks called once after the Bazel invocation completes, with the exit code"),
+    bazel_attempt_end = attr(list[typing.Callable[[TaskContext, int], None]], default = [], description = "Hooks called once after EACH Bazel invocation attempt (including retries), with that attempt's exit code. Runs after the wait() returns and before the retry decision, so a hook can take corrective action (e.g. clean stranded sandbox state) that might let the next retry succeed. Distinct from `build_end`, which fires once after the final attempt."),
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply crates/aspect-cli/src/builtins/aspect/bazel.axl:84]: Addressed
